bundles/org.openhab.binding.sunsynk/src/main/java/org/openhab/binding/sunsynk/internal/handler/SunSynkInverterHandler.java

```diff
@@ -0,0 +1,416 @@
+import java.util.concurrent.TimeUnit;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.openhab.binding.sunsynk.internal.api.DeviceController;
+import org.openhab.binding.sunsynk.internal.api.dto.Battery;
+import org.openhab.binding.sunsynk.internal.api.dto.Daytemps;
+import org.openhab.binding.sunsynk.internal.api.dto.Daytempsreturn;
+import org.openhab.binding.sunsynk.internal.api.dto.Grid;
+import org.openhab.binding.sunsynk.internal.api.dto.RealTimeInData;
+import org.openhab.binding.sunsynk.internal.api.dto.Settings;
+import org.openhab.binding.sunsynk.internal.api.exception.SunSynkAuthenticateException;
+import org.openhab.binding.sunsynk.internal.api.exception.SunSynkGetStatusException;
+import org.openhab.binding.sunsynk.internal.api.exception.SunSynkSendCommandException;
+import org.openhab.binding.sunsynk.internal.config.SunSynkInverterConfig;
+import org.openhab.core.library.types.DecimalType;
+import org.openhab.core.library.types.OnOffType;
+import org.openhab.core.library.types.StringType;
+import org.openhab.core.thing.Bridge;
+import org.openhab.core.thing.ChannelUID;
+import org.openhab.core.thing.Thing;
+import org.openhab.core.thing.ThingStatus;
+import org.openhab.core.thing.ThingStatusDetail;
+import org.openhab.core.thing.binding.BaseThingHandler;
+import org.openhab.core.types.Command;
+import org.openhab.core.types.RefreshType;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link SunSynkInverterHandler} is responsible for handling commands, which are
+ * sent to one of the Inverter channels.
+ *
+ * @author Lee Charlton - Initial contribution
+ */
+
+@NonNullByDefault
+public class SunSynkInverterHandler extends BaseThingHandler {
+
+    private final Logger logger = LoggerFactory.getLogger(SunSynkInverterHandler.class);
+    private @Nullable ZonedDateTime lockoutTimer = null;
+    private DeviceController inverter = new DeviceController();
+    private @Nullable ScheduledFuture<?> refreshTask;
+    private boolean batterySettingsUpdated = false;
+    private SunSynkInverterConfig config = new SunSynkInverterConfig();
+    public @Nullable Settings tempInverterChargeSettings = inverter.tempInverterChargeSettings; // Holds modified
+                                                                                                // battery settings.
+
+    public SunSynkInverterHandler(Thing thing) {
+        super(thing);
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        Optional<SunSynkAccountHandler> checkBridge = getSafeBridge();
+        if (!checkBridge.isPresent()) {
+            logger.debug("Failed to find associated SunSynk Bridge.");
+            return;
+        }
+
+        if (command instanceof RefreshType) {
+            refreshStateAndUpdate();
+        } else {
+            this.tempInverterChargeSettings = inverter.getBatteryChargeSettings();
```

Comment:
create local var to prevent null warnings
```suggestion
            this.tempInverterChargeSettings = inverter.getBatteryChargeSettings();
            Settings tempInverterChargeSettings = this.tempInverterChargeSettings;
            if (tempInverterChargeSettings == null) {
                logger.debug("Settings are null, this is unexpected");
                return;
            }
```

Reply:
OBE